test/integration/filters/add_body_filter.cc

```diff
@@ -24,11 +24,21 @@ class AddBodyStreamFilter : public Http::PassThroughFilter {
       Buffer::OwnedImpl body("body");
       headers.setContentLength(body.length());
       decoder_callbacks_->addDecodedData(body, false);
+    } else {
+      headers.removeContentLength();
```

Comment:
Why would we hit this if this is used with header only requests?

Reply:
header only request doesn't mean the fin bit arrives with the headers, in the case of QUIC.
will comment.